lib/msprime.c

```diff
@@ -4366,55 +4340,72 @@ msp_pedigree_climb(msp_t *self)
 
             /* Merge segments inherited from this ind and recombine */
             // TODO: Make sure population gets properly set when more than one
-            ret = msp_merge_ancestors(self, segments, 0, 0, &merged_segment, parent_id);
+            ret = msp_merge_ancestors(self, segments, 0, 0, &merged_head, parent_id, node);
             if (ret != 0) {
                 goto out;
             }
-            if (merged_segment == NULL) {
+            if (merged_head == NULL) {
                 // This lineage has coalesced
                 continue;
             }
+            ret = msp_flush_edges(self);
```

Comment:
I think we should only have the flush edges at the end of the loop - we're doing it twice here, right?